### PR TITLE
shadow-single-player-owner-fix:

### LINKED ISSUE #1442: problem of command shadow(1.16.5)
It's known that `/player [player's name] shadow` doesn't work in single player mode and the single player server will stop working if you try to make yourself shadow. I tried to use this command in my single player save and I found that the fake player seems to join the game not with the latest playerdata but the playerdata saved last time. When I opened the save again, I found that the playerdata was not saved but the world was updated. For example, I can duplicate any item by following the process below:

1. get the item you want to duplicate in your inventory
2. save and quit to title
3. open the save again
4. drop the item you want to duplicate on the ground
5. type'/player [your name] shadow' and enter
6. you will see 'you are logged in from another location', click 'back to server list' and then back to title screen
7. open the save again, you'll see the item you want to duplicate both in your inventory and on the ground

### PR DESCRIPTION
**Pull request by silnarm**
* Disallow shadowing a single player server owner. Fixes #1442

### CODE REVIEW

File: src/main/java/carpet/commands/PlayerCommand.java [hunk 1]
```diff
@@ -365,6 +366,11 @@ private static int shadow(CommandContext<CommandSourceStack> context)
             Messenger.m(context.getSource(), "r Cannot shadow fake players");
             return 0;
         }
+        MinecraftServer server = Objects.requireNonNull(player.getServer());
```
Comment (altrisi) [inline src/main/java/carpet/commands/PlayerCommand.java:369]: Don't think the server can ever be null for a ServerPlayer.

File: src/main/java/carpet/commands/PlayerCommand.java [hunk 2]
```diff
@@ -365,6 +366,11 @@ private static int shadow(CommandContext<CommandSourceStack> context)
...
+        if (server.isSingleplayer() && server.isSingleplayerOwner(player.getGameProfile())) {
```
Comment (altrisi) [inline src/main/java/carpet/commands/PlayerCommand.java:370]: No need to check `isSinglePlayer`, dedicated server will always return `false` for `isSinglePlayerOwner`.

**Review by altrisi — COMMENTED** [review]
Some tiny comments

File: src/main/java/carpet/commands/PlayerCommand.java (hunk 1 above)
Comment (silnarm) [reply src/main/java/carpet/commands/PlayerCommand.java:369]: Yeah, I've even seen this in action plenty already I think. Too trusting of what IntelliJ is telling me, time to disable some inspections.

**Review by altrisi — APPROVED** [review]

**Comment by altrisi** [comment]
I believe this crashes if the given player isn't online (`getPlayer` returns `null`), will fix it in a small refactor I'm doing to the player command.

**Comment by silnarm** [comment]
Oops, missed a test case there I guess.

**Pull request merged**